### PR TITLE
Add packit support

### PR DESCRIPTION
**Pull request by ianballou**
I used hammer-cli-foreman's packit file as an example.

### CODE REVIEW

**Review by chris1984 — APPROVED** [review]
ACK

**Comment by ianballou** [comment]
https://github.com/theforeman/foreman-packaging/pull/10731 will fix the builds

File: gemfile.d/test.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+group :test do
+  gem 'ci_reporter', '>= 1.6.3', "< 2.0.0"
```
Comment (ekohl) [inline gemfile.d/test.rb:2]: This one was for Jenkins. I don't think it supports GHA

File: Gemfile [hunk 2]
```diff
@@ -7,3 +7,7 @@ gemspec
   local_gemfile = File.join(File.dirname(__FILE__), file_name)
   self.instance_eval(Bundler.read_file(local_gemfile)) if File.exist?(local_gemfile)
 end
+
+Dir[File.join(__dir__, 'gemfile.d', '*.rb')].each do |bundle|
```
Comment (ekohl) [inline Gemfile:11]: There's no need for this here. You can just define it in the Gemfile itself. Gemfile.d is for Foreman plugins.

While we don't use it today, dependabot doesn't support `eval_gemfile` so I'd stick to a plain Gemfile.

Though the instance_eval above already breaks dependabot.

**Review by ekohl — CHANGES_REQUESTED** [review]
I'd also drop the test deps from the gemspec to avoid duplication

**Comment by ianballou** [comment]
This PR should fix our nightly builds, which folks are waiting to test.

File: gemfile.d/test.rb (hunk 1 above)
Comment (ekohl) [reply gemfile.d/test.rb:2]: If you're dropping the use of it, you can also drop the dependency
```suggestion
```

**Comment by ianballou** [comment]
@ekohl thanks for catching the gemfile folder, it's deleted now.

**Comment by ianballou** [comment]
Did a quick test, my PR doesn't seem to break anything in h-c-k at least.

**Review by chris1984 — APPROVED** [review]

**Comment by ianballou** [comment]
Merging since it _should_ help with nightly, but we can make further improvements as needed.

**Pull request merged**